stdlib/public/core/Codable.swift

```diff
@@ -89,6 +89,58 @@ extension CodingKey {
   public var debugDescription: String {
     return description
   }
+
+  /// A simplified description: the int value, if present, in square brackets.
+  /// Otherwise, the string value by itself. Used when concatenating coding keys
+  /// to form a path when printing debug information.
+  /// - parameter isFirst: Whether this is the first key in a coding path, in
+  ///   which case we will omit the prepended '.' delimiter from string keys.
+  func errorPresentationDescription(isFirstInCodingPath isFirst: Bool = true) -> String {
+    if let intValue {
+      return "[\(intValue)]"
+    } else {
+      let delimiter = isFirst ? "" : "."
+      return "\(delimiter)\(stringValue.escapedForCodingKeyErrorPresentationDescription)"
+    }
+  }
+}
+
+private extension [any CodingKey] {
+  /// Concatenates the elements of an array of coding keys and joins them with
+  /// "/" separators to make them read like a path.
+  func errorPresentationDescription() -> String {
```

Comment:
Similarly, in code within the Standard Library, [we expect to put access modifiers on each member of an extension](https://github.com/swiftlang/swift/blob/main/docs/StandardLibraryProgrammersManual.md#explicit-access-modifiers), rather than relying on context.

```suggestion
extension [any CodingKey] {
  /// Concatenates the elements of an array of coding keys and joins them with
  /// "/" separators to make them read like a path.
  private func _errorPresentationDescription() -> String {
```

(Also note the leading underscore!)